src/Entity/Slot.php

```diff
@@ -0,0 +1,223 @@
+ *     "disable" = "/admin/config/content/acquia-lift/slots/{acquia_lift_slot}/disable",
+ *     "enable" = "/admin/config/content/acquia-lift/slots/{acquia_lift_slot}/enable",
+ *   }
+ * )
+ */
+class Slot extends ConfigEntityBase implements SlotInterface {
+
+  /**
+   * The ID of the slot.
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The slot label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * A string describing the slot.
+   *
+   * @var string
+   */
+  protected $description;
+
+  /**
+   * The slot visibility settings
+   *
+   * @var string
+   */
+  protected $visibility;
+
+  /**
+   * The slot CSS Url
+   *
+   * @var string
+   */
+  protected $css_url;
+
+  /**
+   * The Lift API Helper.
+   *
+   * @var \Acquia\LiftClient\Lift
+   */
+  protected $liftClient;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $values, $entity_type) {
+    parent::__construct($values, $entity_type);
+
+    try {
+      /** @var \Drupal\acquia_lift\Service\Helper\LiftAPIHelper $liftHelper */
+      $liftHelper = \Drupal::getContainer()
+        ->get('acquia_lift.service.helper.lift_api_helper');
+      $this->liftClient = $liftHelper->getLiftClient();
+    } catch (AcquiaLiftException $e) {
+      drupal_set_message(t($e->getMessage()), 'error');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    return $this->description;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setDescription($description) {
+    $this->description = $description;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCssUrl() {
+    return $this->css_url;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setCssUrl($cssUrl) {
+    $this->css_url = $cssUrl;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getHtml() {
+    $css_html = "";
+    if (!empty($this->css_url)) {
+      $css_html = " data-lift-css=\"" . $this->css_url . "\"";
+    }
+    return "<div data-lift-slot=\"" . $this->id() . "\"" . $css_html . "/>";
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getVisibility() {
+    if (!empty($this->visibility) && is_array($this->visibility)) {
+      return new Visibility($this->visibility);
+    }
+    return new Visibility([]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setVisibility(Visibility $visibility) {
+    $this->visibility = $visibility->getArrayCopy();
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExternalSlot() {
+    $slot = new \Acquia\LiftClient\Entity\Slot();
+    $slot->setId($this->id());
+    $slot->setLabel($this->label);
+    $slot->setDescription($this->description);
+    $slot->setStatus($this->status());
+    $slot->setHtml($this->getHtml());
+    $slot->setVisibility($this->getVisibility());
+    return $slot;
+  }
```

Comment:
Why not "clone" keyword? "clone" should create a brand new hard copy of the object. Two catches are: 1) original object's referenced objects are not cloned, only reference were cloned. 2) cloned object and services are detached from EntityManager and Containers.

Both catch don't apply so I think there is a good chance here we can use clone.